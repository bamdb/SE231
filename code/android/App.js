import React from "react";
import { View, Text ,Button,TextInput} from "react-native";
import { createMaterialToppTabNavigator,createStackNavigator, createAppContainer } from "react-navigation";
import { createMaterialBottomTabNavigator } from "react-navigation-material-bottom-tabs";
import { Grid, Icon } from '@ant-design/react-native';
import Storage from 'react-native-storage'
import axios from 'axios'
import HomeScreen from './src/loginpage'
import DetailsScreen from './src/browsepage'
import Itemdetail from './src/itemdetail'
import Topicpage from './src/topicpage'
import Activitypage from './src/activity'
import Topicdetail from './src/topicdetail'
import BasicTabBarExample from "./test";
import Testcase from "./test";

const AppNavigator = createMaterialBottomTabNavigator(
  {
    Home: {
      screen: HomeScreen,
      navigationOptions: {
        tabBarIcon: ({ tintColor }) => (
          <Icon name='login' color={tintColor} size={24} />
        )
      },
    },
    Browse: {
      screen:DetailsScreen,
      navigationOptions:{
        tabBarIcon: ({ tintColor }) => (
          <Icon name='table' color={tintColor} size={24} />
        )
      }
    },
    Topic: {
      screen:Topicpage,
      navigationOptions:{
        tabBarIcon: ({ tintColor }) => (
          <Icon name='menu' color={tintColor} size={24} />
        )
      }
    },
    Activity:{
      screen:Activitypage,
      navigationOptions:{
        tabBarIcon: ({ tintColor }) => (
          <Icon name='bell' color={tintColor} size={24} />
        )
      }
    }


  },
  {
    initialRouteName: 'Home',
    shifting:false,
    activeColor: 'purple',
    inactiveColor: 'white',
    barStyle: { backgroundColor: 'pink' },
  },

);

const MainScreenNavigator = createStackNavigator({

  Home: { screen: createAppContainer(AppNavigator) },

  Itemdetail:{screen: Itemdetail},
  Topicdetail:{screen: Topicdetail},


},
{
  initialRouteName: "Home",
    defaultNavigationOptions: {

      header:null,
    },

});
export default createAppContainer(MainScreenNavigator);
