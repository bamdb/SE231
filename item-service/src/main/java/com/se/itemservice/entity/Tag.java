package com.se.itemservice.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

public class Tag {
    private String tagname;

    private List<Long> userList;

    public String getTagname() {
        return tagname;
    }

    public void setTagname(String tagname) {
        this.tagname = tagname;
    }

    public List<Long> getUserList() {
        return userList;
    }

    public void setUserList(List<Long> userList) {
        this.userList = userList;
    }
}
