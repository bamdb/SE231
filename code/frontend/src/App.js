import React from 'react';
import logo from './logo.svg';
import './App.css';
import Search from './component/search'
import Scheduletable from'./component/scheduletable'
import Item from './component/item'
import Userinfo from './component/userinfo'
import Homepage from './component/homepage'
import Navigation from "./component/navigation";
import TopItemList from "./component/topitemlist";
import Browserlist from "./component/browserlist";

import {
  HashRouter as Router,
  Route,
  Link,
  Redirect,
  withRouter
} from "react-router-dom";
import UserPage from "./component/userpage";
function App() {
  return (
    <div >

      <Router >
        <Route path={'/browserlist'} component={Browserlist}></Route>
        <Route path={'/topitemlist'} component={TopItemList}></Route>
        <Route path={'/navigate'} component={Navigation}></Route>
      <Route path={'/search'} component={Search}></Route>
        <Route path={'/homepage'} component={Homepage}></Route>
      <Route path={'/schedule'} component={Scheduletable}></Route>
        <Route path={'/item'} component={Item}></Route>
          <Route path={'/userinfo'} component={Userinfo}></Route>
          <Route path={'/userpage'} component={UserPage}></Route>

      </Router>
    </div>
  );
}

export default App;
