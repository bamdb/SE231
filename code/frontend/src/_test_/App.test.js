import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';

import Tag from '../component/tag'
import TestRenderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('test tag', () => {
  const testrenderer=TestRenderer.create(<Tag/>);
  const testinstance=testrenderer.root;
  let tree = testrenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

