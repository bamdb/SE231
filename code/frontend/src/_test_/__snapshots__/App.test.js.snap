// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tag 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    id="tagmain"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto"
      >
        <div
          className="MuiChip-root MuiChip-colorSecondary MuiChip-clickableColorSecondary MuiChip-clickable"
          id="chip"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiChip-label"
          >
            搞笑
          </span>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto"
      >
        <div
          className="MuiChip-root MuiChip-colorSecondary MuiChip-clickableColorSecondary MuiChip-clickable"
          id="chip"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiChip-label"
          >
            热血
          </span>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto"
      >
        <div
          className="MuiChip-root MuiChip-colorSecondary MuiChip-clickableColorSecondary MuiChip-clickable"
          id="chip"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiChip-label"
          >
            王道
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
